### ISSUE TITLE
Rancher pod stuck in crashLoop in Rancherd setup 

### ISSUE THREAD

**Issue by MonzElmasry** [rancher/rancher#29051]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- Run rancherd build script with rancher/rancher:master-head image in rancherd chart https://github.com/rancher/rancher/blob/master/cmd/rancherd/chart/values.yaml#L47
- Docker build the runtime image
- run rancherd server

**Result:**
RKE2 is up and running. Node is ready, All control plan components are running  
Rancher pod will get stuck in crashloop status , viewing the pod logs : 
```
2020/09/21 18:31:35 [INFO] Rancher startup complete
2020/09/21 18:31:35 [WARNING] Cluster local has no creatorId annotation. Cannot create Default project
2020/09/21 18:31:35 [WARNING] Cluster local has no creatorId annotation. Cannot create System project
E0921 18:31:35.359940       8 runtime.go:78] Observed a panic: "assignment to entry in nil map" (assignment to entry in nil map)
goroutine 1159 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x3bf9de0, 0x4cf34a0)
	/go/pkg/mod/github.com/rancher/apimachinery@v0.19.0-rancher1/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/github.com/rancher/apimachinery@v0.19.0-rancher1/pkg/util/runtime/runtime.go:48 +0x82
panic(0x3bf9de0, 0x4cf34a0)
	/usr/local/go/src/runtime/panic.go:969 +0x166
github.com/rancher/rancher/pkg/controllers/management/auth.(*mgr).addRTAnnotation(0xc0028e6480, 0x4de9200, 0xc00311f100, 0x44ad5ec, 0x7, 0x4de9200, 0xc00311f100, 0xc00129d200, 0xc00129d290)
	/go/src/github.com/rancher/rancher/pkg/controllers/management/auth/project_cluster_handler.go:560 +0x40d
github.com/rancher/rancher/pkg/controllers/management/auth.(*clusterLifecycle).sync(0xc003b58a58, 0xc001107af0, 0x5, 0xc0013c4300, 0xc0031fae00, 0x7f927201f7d0, 0xc001900480, 0x0)
	/go/src/github.com/rancher/rancher/pkg/controllers/management/auth/project_cluster_handler.go:177 +0x1a8
github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3.(*clusterController).AddHandler.func1(0xc001107af0, 0x5, 0x4411940, 0xc0013c4300, 0xc00194db20, 0x3, 0x3, 0x0)
	/go/src/github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3/zz_generated_cluster_controller.go:155 +0x63
github.com/rancher/norman/controller.(*genericController).AddHandler.func1(0xc001107af0, 0x5, 0x4de9200, 0xc0013c4300, 0x4de9200, 0xc0013c4300, 0x0, 0x0)
	/go/pkg/mod/github.com/rancher/norman@v0.0.0-20200908202416-992a35eef40f/controller/generic_controller.go:62 +0x221
github.com/rancher/lasso/pkg/controller.SharedControllerHandlerFunc.OnChange(0xc003e2fb30, 0xc001107af0, 0x5, 0x4de9200, 0xc0013c4300, 0x4de9200, 0xc0013c4300, 0x0, 0x0)
	/go/pkg/mod/github.com/rancher/lasso@v0.0.0-20200905045615-7fcb07d6a20b/pkg/controller/sharedcontroller.go:29 +0x4e
github.com/rancher/lasso/pkg/controller.(*sharedHandler).OnChange(0xc000d5cea0, 0xc001107af0, 0x5, 0x4de9200, 0xc0013c4300, 0xc001011301, 0x0)
	/go/pkg/mod/github.com/rancher/lasso@v0.0.0-20200905045615-7fcb07d6a20b/pkg/controller/sharedhandler.go:65 +0x14a
github.com/rancher/lasso/pkg/controller.(*controller).syncHandler(0xc000932840, 0xc001107af0, 0x5, 0xc00145e810, 0xc00194dd50)
	/go/pkg/mod/github.com/rancher/lasso@v0.0.0-20200905045615-7fcb07d6a20b/pkg/controller/controller.go:210 +0xd1
github.com/rancher/lasso/pkg/controller.(*controller).processSingleItem(0xc000932840, 0x38f1ce0, 0xc004ac8570, 0x0, 0x0)
	/go/pkg/mod/github.com/rancher/lasso@v0.0.0-20200905045615-7fcb07d6a20b/pkg/controller/controller.go:192 +0xd7
github.com/rancher/lasso/pkg/controller.(*controller).processNextWorkItem(0xc000932840, 0x203000)
	/go/pkg/mod/github.com/rancher/lasso@v0.0.0-20200905045615-7fcb07d6a20b/pkg/controller/controller.go:169 +0x51
github.com/rancher/lasso/pkg/controller.(*controller).runWorker(0xc000932840)
	/go/pkg/mod/github.com/rancher/lasso@v0.0.0-20200905045615-7fcb07d6a20b/pkg/controller/controller.go:158 +0x2b
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc00034f950)
	/go/pkg/mod/github.com/rancher/apimachinery@v0.19.0-rancher1/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc00034f950, 0x4dbf1a0, 0xc00145ec90, 0xc0006ad801, 0xc00116f620)
	/go/pkg/mod/github.com/rancher/apimachinery@v0.19.0-rancher1/pkg/util/wait/wait.go:156 +0xa3
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00034f950, 0x3b9aca00, 0x0, 0x1, 0xc00116f620)
	/go/pkg/mod/github.com/rancher/apimachinery@v0.19.0-rancher1/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc00034f950, 0x3b9aca00, 0xc00116f620)
	/go/pkg/mod/github.com/rancher/apimachinery@v0.19.0-rancher1/pkg/util/wait/wait.go:90 +0x4d
created by github.com/rancher/lasso/pkg/controller.(*controller).run
	/go/pkg/mod/github.com/rancher/lasso@v0.0.0-20200905045615-7fcb07d6a20b/pkg/controller/controller.go:129 +0x31f
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map

```

**Comment by dweomer** [comment]
Looks like a bug in rancher/rancher: https://github.com/rancher/rancher/blob/master/pkg/controllers/management/auth/project_cluster_handler.go#L560

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rancher/blob/6f19189d541d0ad0a830b9d82f8de8950596f1d7/pkg/data/dashboard/cluster_data.go#L37-L38 when creating local cluster, we append local-admin-user to the annotation, and rancher creates system and default project based on the annotation. So we might have to fix the panic but also add the annotation into cluster so the existing logic will not be broken. Adding the annotation logic can be done in `reset-admin` command.

**Comment by davidnuzik** [comment]
Added blocker label. Based on https://github.com/rancher/rke2/issues/201#issuecomment-696332401 it sounds like this is blocking getting https://github.com/rancher/rke2/issues/201 done, which is needed for GA.

**Comment by rancher-max** [comment]
#### Validated this is fixed in latest rc: `v2.5.0-rc1`

I am able to get a multinode rancherd cluster running with rancher running successfully by using the binary provided in the release artifacts. My steps are (node 1):

```
wget -O rancherd https://github.com/rancher/rancher/releases/download/v2.5.0-rc1/rancherd-amd64 && \
sudo mv rancherd /usr/local/bin/rancherd && \
sudo chmod +x /usr/local/bin/rancherd

sudo rancherd server --write-kubeconfig-mode 644 2>&1 &
```
The other nodes followed the same steps but included the `--token` and `--server flags`.

I also did this using `--tls-san` flag for all nodes.
